### DISCUSSION TITLE
Inline form editing not working

### DISCUSSION THREAD

**Discussion by qweluke** [react-hook-form/react-hook-form#5282] [Q&A] [answered]
Hello, I'm trygint to make some inline form editing but for some reason it is not working :/
https://codesandbox.io/s/react-hook-form-conditional-forked-6hunc?file=/src/index.js

When you click on "edit" button the text fields are empty and if you change text to anything, then click "save' - form state is not updated.
What am I doing wrong?

**Comment by thanh-nguyen-95** [comment]
Everything works fine for me. Is it a correct CodeSandBox url?

**Reply by qweluke** [reply]
@thanh-nguyen-95 i fixed in partially in the meantime, i'm still having weird issue when deleting rows.

ie:
edit row 0 - change firstName to "AAAAAAAA" and click save
add another row, set firstName to "BBBBBBBB" and click save
delete row with firstName eq "BBBBBBBB" - row 0 will change firstName from "AAAAAAAA" to "Bill" but then if you click "add another row" it will switch back to "AAAAAAAA"

**Reply by qweluke** [reply]
I've recorded some short video to demonstrate this problem
https://user-images.githubusercontent.com/193701/119276694-ccbcdb80-bc1b-11eb-91f5-930e69dfd569.mov

**Comment by thanh-nguyen-95** [comment, marked as answer]
Changing the code like below works for me. The problem is the `as` prop. It needs to be an `input` for `react-hook-form` (RHF) to work since RHF get/set value using input dom element.

If `<input readOnly={true} />` does not meet your needs, I suggesting upgrading to RHF v7 (you're running v5.7.2)

```jsx
return (
  <form onSubmit={handleSubmit(onSubmit)}>
    <ul>
      {fields.map((item, index) => {
        const isEditing = index === editRow

        return (
          <li key={item.id}>
            <Controller
              as={isEditing ? <input /> : <input readOnly={true} />}                
              control={control}
              name={`test[${index}].firstName`}
              defaultValue={item.firstName}
            />

            <Controller
              as={isEditing ? <input /> : <input readOnly={true} />} 
              control={control}
              name={`test[${index}].lastName`}
              defaultValue={item.lastName}
            />

            <button type="button" onClick={() => remove(index)}>
              Delete
            </button>

            <button
              type="button"
              onClick={() => setEditRow(isEditing ? null : index)}
            >
              {isEditing  ? "SAVE" : "EDIT"}
            </button>
          </li>
        );
      })}
    </ul>

    <button
      type="button"
      onClick={() => append({ firstName: "XX", lastName: "YY" })}
    >
      ADD ANOTHER ROW
    </button>

    <input type="submit" />
  </form>
);
```

Edit: fix typos

**Reply by bluebill1049** [reply]
Thank you :)
